From cda5710b8d071825956e47bc7cccbd2ca4e9452a Mon Sep 17 00:00:00 2001
From: William Jordan <bjordan@habana.ai>
Date: Fri, 16 May 2025 21:47:31 +0300
Subject: [PATCH] [SW-218976] add entity aux names to agents

Add Entity AUX Name PDRs to the agent PLDM
PDR database.

Change-Id: I30010f4198347d1e73af9cd26f91f21ff46bc24d
Signed-off-by: William Jordan <bjordan@habana.ai>
---

diff --git a/common/include/gaudi3/pldm_pdr.h b/common/include/gaudi3/pldm_pdr.h
index 83199de..53cda55 100644
--- a/common/include/gaudi3/pldm_pdr.h
+++ b/common/include/gaudi3/pldm_pdr.h
@@ -34,6 +34,9 @@
 	PLDM_RH_SENSOR_ETH_SERDES_AVAILABILITY_OAM			= 3106,
 	PLDM_RH_SENSOR_ETH_SERDES_AVAILABILITY_RETIMER_1		= 3107,
 	PLDM_RH_SENSOR_ETH_SERDES_AVAILABILITY_RETIMER_2		= 3108,
+	PLDM_RH_ENTITY_AUX_NAMES_ADDIN_CARD				= 4000,
+	PLDM_RH_ENTITY_AUX_NAMES_OAM_CPLD				= 4001,
+	PLDM_RH_ENTITY_AUX_NAMES_ACCELERATOR				= 4002,
 	PLDM_RH_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_HIGH			= 4011,
 	PLDM_RH_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_LOW			= 4012,
 	PLDM_RH_SENSOR_AUX_NAMES_DEVICE_PDR_REPOSITORY_VERSION		= 4020,
@@ -136,6 +139,12 @@
 	PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_VENDOR_ID,
 };
 
+enum pldm_entity_aux_ridx {
+	PLDM_RIDX_ENTITY_AUX_NAMES_ADDIN_CARD,
+	PLDM_RIDX_ENTITY_AUX_NAMES_OAM_CPLD,
+	PLDM_RIDX_ENTITY_AUX_NAMES_ACCELERATOR,
+};
+
 enum pldm_oem_state_set_ridx {
 	PLDM_RIDX_SENSOR_OEM_STATE_SET_FW_STATE,
 	PLDM_RIDX_SENSOR_OEM_STATE_SET_PPBOOT_IMAGE_SOURCE,
@@ -157,6 +166,9 @@
 extern struct pldm_pdr_name_record pldm_pdr_name_records[];
 extern const size_t pldm_pdr_name_records_length;
 
+extern struct pldm_pdr_entity_name_record pldm_pdr_entity_name_records[];
+extern const size_t pldm_pdr_entity_name_records_length;
+
 extern struct pldm_pdr_oem_state_set_record pldm_pdr_oem_state_set_records[];
 extern const size_t pldm_pdr_oem_state_set_records_length;
 
diff --git a/common/src/gaudi3/ras/pldm/pldm_pdr.c b/common/src/gaudi3/ras/pldm/pldm_pdr.c
index 354c3cb..da82763 100644
--- a/common/src/gaudi3/ras/pldm/pldm_pdr.c
+++ b/common/src/gaudi3/ras/pldm/pldm_pdr.c
@@ -890,6 +890,18 @@
  *  - Replace ' ' with '_'
  *  - Remove '.'
  */
+const uint16_t oamAddInCardAccelerator_str[] = {
+	0x6f00, 0x6100, 0x6d00, 0x4100, 0x6400, 0x6400, 0x4900,
+	0x6e00, 0x4300, 0x6100, 0x7200, 0x6400, 0x0000
+};
+const uint16_t oamCPLD_str[] = {
+	0x6f00, 0x6100, 0x6d00, 0x4400, 0x5000, 0x4c00, 0x4400,
+	0x0000
+};
+const uint16_t accelerator_str[] = {
+	0x6100, 0x6300, 0x6300, 0x6500, 0x6c00, 0x6500, 0x7200,
+	0x6100, 0x7400, 0x6f00, 0x7200, 0x0000
+};
 const uint16_t asicReset_str[] = {
 	0x6100, 0x7300, 0x6900, 0x6300, 0x5200, 0x6500, 0x7300,
 	0x6500, 0x7400, 0x0000,
@@ -1780,6 +1792,62 @@
 
 const size_t pldm_pdr_name_records_length = ARRAY_SIZE(pldm_pdr_name_records);
 
+struct pldm_pdr_entity_name_record pldm_pdr_entity_name_records[] = {
+	[PLDM_RIDX_ENTITY_AUX_NAMES_ADDIN_CARD] = { .entity_aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_ENTITY_AUX_NAMES_ADDIN_CARD,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_ENTITY_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.entity_type = PLDM_ENTITY_TYPE_ADDIN_CARD,
+		.entity_instance = 1,
+		.entity_container_id = PLDM_CONTAINER_ADDIN_CARD,
+		.shared_name_count = 0,
+		.name = {
+			.name_string_count = 1,
+			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+			.name = oamAddInCardAccelerator_str,
+		},
+	}},
+	[PLDM_RIDX_ENTITY_AUX_NAMES_OAM_CPLD] = { .entity_aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_ENTITY_AUX_NAMES_OAM_CPLD,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_ENTITY_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.entity_type = PLDM_ENTITY_TYPE_CPLD,
+		.entity_instance = 1,
+		.entity_container_id = PLDM_CONTAINER_ADDIN_CARD,
+		.shared_name_count = 0,
+		.name = {
+			.name_string_count = 1,
+			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+			.name = oamCPLD_str,
+		},
+	}},
+	[PLDM_RIDX_ENTITY_AUX_NAMES_ACCELERATOR] = { .entity_aux_names = {
+		.header = {
+			.record_handle = PLDM_RH_ENTITY_AUX_NAMES_ACCELERATOR,
+			.header_version = PLDM_PDR_HEADER_VERSION,
+			.type = PLDM_PDR_TYPE_ENTITY_AUX_NAMES,
+			.record_change_number = 0,
+		},
+		.entity_type = PLDM_ENTITY_TYPE_ACCELERATOR,
+		.entity_instance = 1,
+		.entity_container_id = PLDM_CONTAINER_ADDIN_CARD,
+		.shared_name_count = 0,
+		.name = {
+			.name_string_count = 1,
+			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+			.name = accelerator_str,
+		},
+	}},
+};
+
+const size_t pldm_pdr_entity_name_records_length = ARRAY_SIZE(pldm_pdr_entity_name_records);
+
 struct pldm_pdr_oem_state_set_record pldm_pdr_oem_state_set_records[] = {
 	[PLDM_RIDX_SENSOR_OEM_STATE_SET_FW_STATE] = { .oem_state_set = {
 		.header = {
diff --git a/common/src/ras/pldm/pldm.c b/common/src/ras/pldm/pldm.c
index 1c735ec..fc84d2d 100644
--- a/common/src/ras/pldm/pldm.c
+++ b/common/src/ras/pldm/pldm.c
@@ -394,6 +394,8 @@
 		.num_records = pldm_pdr_records_length,
 		.name_records = pldm_pdr_name_records,
 		.num_name_records = pldm_pdr_name_records_length,
+		.entity_name_records = pldm_pdr_entity_name_records,
+		.num_entity_name_records = pldm_pdr_entity_name_records_length,
 		.oem_state_set_records = pldm_pdr_oem_state_set_records,
 		.num_oem_state_set_records = pldm_pdr_oem_state_set_records_length,
 		.oem_unit_records = pldm_pdr_oem_unit_records,
